examples/bat_gen/bat_fit.cxx

```diff
@@ -73,10 +73,12 @@ bat_fit::bat_fit(std::string name, std::unique_ptr<yap::Model> M, const std::vec
     //                         [](int n, const yap::IntegralMap::value_type& v)
     //                         {return n + v.second.decayTrees().size();});
     // if (N > 1) {
+    unsigned i(0); // running index to avoid name collisions
     for (const auto& b2_dtvi : Integral_.integrals())
         for (const auto& dt : b2_dtvi.second.decayTrees()) {
             DecayTrees_.push_back(dt);
-            AddObservable("fit_frac(" + to_string(*dt->freeAmplitude()->decayChannel()) + " M = " + yap::spin_to_string(dt->freeAmplitude()->twoM()) + ")", 0, 1.1);
+            AddObservable("fit_frac(" + to_string(*dt->freeAmplitude()) + ")" + std::to_string(i++), 0, 1.1);
```

Comment:
When there are two decay trees which only differ by some daughters' spin projections, e.g.
`fit_frac(rho0 rho0, M = 0, 0 -> 1 + 1 with L = 1 and S = 1;     d[0] --> pi+ pi-, M = -1, 1 -> 0 + 0 with L = 1 and S = 0 [fixed];     d[1] --> pi+ pi-, M = -1, 1 -> 0 + 0 with L = 1 and S = 0 [fixed])`
and
`fit_frac(rho0 rho0, M = 0, 0 -> 1 + 1 with L = 1 and S = 1;     d[0] --> pi+ pi-, M = 1, 1 -> 0 + 0 with L = 1 and S = 0 [fixed];     d[1] --> pi+ pi-, M = 1, 1 -> 0 + 0 with L = 1 and S = 0 [fixed])`,
the minus signs will be omitted in BAT's safe name, and there is still a name collision:
`Error   : BCVariableSet::Add : Variable with safe name fit_fracrho0rho0M0011withL1andS1d0pipiM1100withL1andS0fixedd1pipiM1100withL1andS0fixed%s exists already.`